packages/marko-web-inquiry/routers/submit.js

```diff
@@ -49,14 +49,15 @@ module.exports = ({ queryFragment, notification, confirmation }) => asyncRoute(a
   };
 
   await Promise.all([
+    // validate google reCaptcha
+    await validateRecaptcha(payload),
     // Store the submission
     storeInquiry({
```

Comment:
Unless you explicitly want to store the submission, you need to wait for the validation to pass _before_ storing it. `Promise.all` will execute all promises at the same time -- moving it up in the list won't cause it to wait.
```suggestion
  // validate google reCaptcha
  await validateRecaptcha(payload);

  await Promise.all([
    // Store the submission
    storeInquiry({
```